app/javascript/src/components/Dashboard/Notes/index.jsx

```diff
@@ -1,46 +1,99 @@
 import React, { useState, useEffect } from "react";
 
+import dayjs from "dayjs";
 import EmptyNotesListImage from "images/EmptyNotesList";
-import { Alert, Button } from "neetoui";
+import { Alert, Button, Toastr } from "neetoui";
 import { Container, Header } from "neetoui/layouts";
 
 import EmptyState from "components/Common/EmptyState";
 import SideMenuBar from "components/Common/SideMenuBar";
-import Toastr from "components/Common/Toastr";
-import { notes as noteList } from "data";
 
 import Card from "./Card";
-import { NOTES_MENU_ITEMS } from "./constants";
+import {
+  NOTES,
+  NOTES_FORM_INITIAL_FORM_VALUES,
+  NOTES_MENU_ITEMS,
+} from "./constants";
+import CreateOrEditPane from "./Pane/CreateOrEdit";
+import { formatNoteDataForApi, formatNoteDataForForm } from "./utils";
 
 const Notes = () => {
   const [searchTerm, setSearchTerm] = useState("");
   const [notes, setNotes] = useState([]);
-  const [selectedNoteForDelete, setSelectedNoteForDelete] = useState({});
+  const [selectedNoteForEditOrDelete, setSelectedNoteForEditOrDelete] =
+    useState({});
   const [showSideMenuBar, setShowSideMenuBar] = useState(true);
   const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [showCreateOrEditPane, setShowCreateOrEditPane] = useState(false);
+  const [isEditNote, setIsEditNote] = useState(false);
 
   useEffect(() => {
-    setNotes(noteList);
+    setNotes(NOTES);
   }, []);
```

Comment:
Please put up a video and explain to me the need for this `useEffect` statement and how your application would behave differently if the useEffect was removed.